### PR TITLE
Add docs for afterMarshal event.

### PR DESCRIPTION
**Pull request by markstory**
Refs cakephp/cakephp#14383

### CODE REVIEW

File: en/orm/saving-data.rst [hunk 1]
```diff
@@ -636,6 +636,34 @@ Moreover, the data in ``beforeMarshal`` is a copy of the passed data. This is
...
+    // In a table or behavior class
+    public function afterMarshal(
+        EventInterface $event,
+        EntityInterface $entity,
+        ArrayObject $data,
+        ArrayObject $options
+    ) {
+        // Don't accept people who have a name starting with J on the 20th
+        // of each month.
+        if (substr($entity->name, 1) == 'J' && date('d') === 20) {
```
Comment (dereuromark) [inline en/orm/saving-data.rst:660]: I fixed up the example in https://github.com/cakephp/docs/commit/67181ea4e30d7c297b6a95693234d2b370e3968f#diff-630ee5f14cae91a8c66d6943b6ce9bfdR660

Comment (markstory) [reply en/orm/saving-data.rst:660]: Thanks!

**Comment by saeideng** [comment]
LGTM 
only I think better to mention that `$data` and `$options` are unchangeable , here or in doc block
and better to  show example between 2 field that on validation process it is impossible

```php
if ($entity->name, 1) == 'J' && $entity->xyz === '123')
         $entity->setError('name', '......');

//or 
if ($entity->x === 1)
     $entity->setError('y', '....message.....');
```
though current example and details are fine :+1:

**Comment by dereuromark** [comment]
I am not sure if we should really keep the $data
The main interest here is the entity itself and its fields. And that you can issue error messages and alike.
The data I do not see much use for, and it makes this callback different than all others (having 1 more arg).

File: en/orm/saving-data.rst (hunk 1 above)
Comment (garas) [reply en/orm/saving-data.rst:660]: `date('d')` is `string` .

Use either `idate('d')` or compare to `'20'`

Comment (dereuromark) [reply en/orm/saving-data.rst:660]: (int) cast would also suffice, I fix that up.

**Pull request merged**